src/Quadratures/Quadratures.jl

```diff
@@ -188,6 +188,29 @@ function interpolation_matrix(
     return SMatrix(M)
 end
 
+function interpolation_matrix(points_to::Vector, points_from)
```

Comment:
How is this different from the one above?

Reply:
Good question. Initially, I just converted from SVectors as you are suggesting, but found that it would take minutes to allocate all the SVectors needed. So, I just skipped all the SVector allocations because at the end I just wanted (Cu/)Arrays.

I changed the code a little bit in the meantime and maybe I reduced the number of SVector allocations needed. I will check again if we need a separate method that doesn't go through SVectors.

Reply:
`SVectors` should generally be more performant, and they're GPU-compatible, so I'd like to understand why the ordinary array version is needed

Reply:
It is needed because allocating lots/large `SVector`s takes time. It takes 26 seconds to prepare a Remapper with 101x101x101 target points. 0.03 seconds if I skip the `SVector`.

Reply:
I understand that, but why would we need a large number of points? The idea is that this function is used point/element-wise. Doing so with SVectors allows it to be efficient and stack allocated.

> 101x101x101 target points

What are the dims supposed to be here?

Reply:
That's the target (lat, lon, z) grid (`z` does't matter because this is only for the horizontal interpolation.) The function `interpolation_matrix` is not element-wise, that's why you have a `points_to`, `points_from`. I mean, I could use it 101x101 times with `points_to` one at the time, but I don't think that's the intent.

Reply:
> I mean, I could use it 101x101 times with `points_to` one at the time, but I don't think that's the intent.

I could be wrong, but I think that's how this function is used elsewhere. Would it be difficult to change to use it the same way?

Reply:
Besides interpolation, this function is used only to interpolate from a quadrature to a different quadrature (so, with multiple N points in both directions). It is true that the old interpolation system used this function pointwise, but this was because the entire interpolation routines were written to be pointwise. 

I want to handle all the interpolation points at the same time, instead of one point at the time. This is one of the key ingredients needed to achieve the performance improvement. So, I need a data structure that holds all the interpolation matrices. If I have to treat each point separately, I would probably have to launch a seperate GPU kernel per point (~ 10^4 kernel launches) or pass ~10^4 SVectors to a single kernel. Instead, I put everything into a CuArray and pass it.

Reply:
We discussed this offline.

The current design is an improvement over what we have. In the long run, it'd probably be better to fuse these interpolations into a single kernel (and just do the interpolation matrix on the fly), but we can leave this for the future.